logging/zap/grpclogger.go

```diff
@@ -44,3 +46,103 @@ func (l *zapGrpcLogger) Printf(format string, args ...interface{}) {
 func (l *zapGrpcLogger) Println(args ...interface{}) {
 	l.logger.Info(fmt.Sprint(args...))
 }
+
+// ReplaceGrpcLoggerV2 replaces the grpc_log.LoggerV2 with the input zap.Logger
+// This logger adheres to the grpc go environment variables GRPC_GO_LOG_VERBOSITY_LEVEL and GRPC_GO_LOG_SEVERITY_LEVEL.
+func ReplaceGrpcLoggerV2(logger *zap.Logger) {
+	zgl := &zapGrpcLoggerV2{}
+	verbosity := os.Getenv("GRPC_GO_LOG_VERBOSITY_LEVEL")
+	if v, err := strconv.Atoi(verbosity); err == nil {
+		zgl.verbosity = v
+	}
+
+	logLevel := os.Getenv("GRPC_GO_LOG_SEVERITY_LEVEL")
+	switch logLevel {
+	case "", "ERROR", "error": // If env is unset, set level to ERROR.
+		zgl.severity = errorLevel
+	case "WARNING", "warning":
+		zgl.severity = warnLevel
+	case "INFO", "info":
+		zgl.severity = infoLevel
+	}
+
+	zgl.Logger = logger.With(zap.String("system", "grpc"), zap.Bool("grpc_log", true))
```

Comment:
can we reuse `SystemField` here as above?

Reply:
right - should have done that.